spec/controllers/spree/admin/base_controller_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Spree::Admin::BaseController, type: :controller do
+  describe '#unauthorized_redirect' do
+    controller(described_class) do
+      def index; authorize!(:read, :something); end
+    end
+
+    context "when user is logged in" do
+      before { sign_in(create(:user)) }
+
+      it "redirects to unauthorized path" do
```

Comment:
Can we add a "when http_referer is not present" scope here, just to make the test conditions explicit? Also, I think all scopes should start with "when", so that you can create a full sentence by reading the scope and test name.